### DISCUSSION TITLE
Loco Deck on different robot than a Crazyflie

### DISCUSSION THREAD

**Discussion by rmdPurdue** [bitcraze/discussions#1719] [Q&A]
Hello,

I have been working with a group of students to explore applying autonomous ground vehicle technologies to moving scenery for theatre use. We stumbled on the Crazyflie drones--which are super cool! We bought a couple, along with the parts for the Loco positioning system to explore indoor absolute positioning. We now wonder if there's any guidance available for communicating with the deck over SPI separate from the Crazyflie drone. We started digging through the documentation, but wondered if you might be able to point us in the right direction, if it is possible to do?

**Comment by ToveRumar** [comment]
Hi @rmdPurdue!

This sounds like a really cool endeavor! 

Unfortunately there is no documentation on how to use the deck outside the Crazyflie. The Loco note does only do a small part of the computation and a lot is done withing the Crazyflie itself, since this has a much more powerful CPU.  
You can get range data from the deck using SPI. This is not something we have tried so there is no documentation, that I know of.  
However, this is exactly what is done in the the driver for the deck that is running on the Crazyflie, so using this as your guide should help you figuring out what you need to do and how. 

The driver can be found here https://github.com/bitcraze/crazyflie-firmware/blob/master/src/deck/drivers/src/locodeck.c along with algorithms for TWR,TDOA (See c files in this directory with corresponding names). 
There is documentation on the LPP packet structure here; https://www.bitcraze.io/documentation/repository/lps-node-firmware/master/protocols/lpp-short-packets-protocol/ and  Loco positioning protocol here; https://www.bitcraze.io/documentation/repository/lps-node-firmware/master/protocols/lpp/ 


Another solution is to put a crazyflie or a Bolt with an attached loco deck on the ground vehicle, and communicate with it over uart to get the absolute positioning. This solution will be more straight forward but of course a bit less flexible. 
My collegue has actually just done something very similar as a project, and you can read about it here; https://www.bitcraze.io/2024/12/dr-kims-last-fun-friday-project/. In this case just switch out the positioning system to LPS, and it should work the same way. 

I hope this helps you find a good solution! It would be fun to hear how it turned out if you try it out!
